### PR TITLE
close 3013 for public for api gateway 164955096

### PR DESCRIPTION
**Pull request by michalbagrowski**
pt: https://www.pivotaltracker.com/story/show/164955096

### CODE REVIEW

File: ansible/group_vars/all/vars.yml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 datadog_enabled: no
+api_gateway: no
```
Comment (dincho) [inline ansible/group_vars/all/vars.yml]: A better variable name would be "local_health_check". Also what about using vars/aeternity/* ? I hope we'll remove ansible group vars usage soon, it does not scale well for our usage.

File: ansible/deploy.yml [hunk 2]
```diff
@@ -239,15 +239,24 @@
...
+        when: health_check and not api_gateway
         connection: local
         tags: [health-check]
 
       - name: API health check
         uri:
           url: "{{ api_base_uri }}/blocks/top"
           timeout: 30
-        when: health_check
+        when: health_check and not api_gateway
```
Comment (dincho) [inline ansible/deploy.yml]: what if you set `health_check: false` for `api-main` instead of this additional check?

File: ansible/deploy.yml [hunk 3]
```diff
@@ -239,15 +239,24 @@
...
+        connection: local
+        tags: [health-check]
+
+      - name: Wait aeternity node API to boot gateway
+        wait_for:
+          port: "{{ node_config.http.external.port }}"
+          host: "localhost"
+          timeout: 300
+        when: api_gateway
         connection: local
```
Comment (dincho) [inline ansible/deploy.yml]: Are you sure `connection: local` works with host:localhost ? I think that this task will be run locally on the controller host and do a request to the controller host

File: ansible/deploy.yml (hunk 2 above)
Comment (michalbagrowski) [reply ansible/deploy.yml]: that was my orginal idea, but you told me to have additional variable.

Comment (dincho) [reply ansible/deploy.yml]: I guess it was misunderstanding  👍

**Pull request merged**